PWGHF/D2H/TableProducer/dataCreatorCharmResoReduced.cxx

```diff
@@ -460,11 +485,117 @@ struct HfDataCreatorCharmResoReduced {
+    float motherPt{-1.f};
+    bool fillHisto{false};
+
+    if constexpr (decChannel == DecayChannel::DstarV0) {
+      // Ds1 → D* K0 → (D0 π+) K0s → ((K-π+) π+)(π+π-)
+      auto indexRec = RecoDecay::getMatchedMCRec(particlesMc, std::array{vecDaughtersReso[0], vecDaughtersReso[1], vecDaughtersReso[2], vecDaughtersReso[3], vecDaughtersReso[4]}, Pdg::kDS1, std::array{+kPiPlus, -kKPlus, +kPiPlus,  +kPiPlus, -kPiPlus}, true, &sign, 3);
+      if (indexRec > -1) {
+        // D* → π- K+ π-
+        indexRec = RecoDecay::getMatchedMCRec(particlesMc, std::array{vecDaughtersReso[0], vecDaughtersReso[1], vecDaughtersReso[2]}, Pdg::kDStar, std::array{-kKPlus, +kPiPlus, +kPiPlus}, true, &signDStar, 2);
+        auto indexRecD0 = RecoDecay::getMatchedMCRec(particlesMc, std::array{vecDaughtersReso[0], vecDaughtersReso[1]}, Pdg::kD0, std::array{+kPiPlus, -kKPlus}, true, &signD0, 1);
+        auto indexRecK0 = RecoDecay::getMatchedMCRec(particlesMc, std::array{vecDaughtersReso[3], vecDaughtersReso[4]}, kK0 , std::array{+kPiPlus, -kPiPlus}, true, &signV0, 2);
+        if (indexRec > -1 && indexRecD0 > -1 && indexRecK0 > -1) {
+          flag = sign * BIT(DecayTypeMc::Ds1ToDStarK0ToD0PiK0s);
+        } // move the following in D+V0 channel
+        else {
+          if (indexRec <= -1){
+            debug = 1;
+            LOGF(debug, "DS1 decays in the expected final state but Dstar does not!");
+          }
+          if (indexRec > -1 && indexRecD0 <= -1){
+            debug = 2;
+            LOGF(debug, "DS1 and Dstar decays in the expected final state but D0 does not!");
+            }
+          if (indexRec > -1 && indexRecK0 <= -1){
+            debug = 3;
+            LOGF(debug, "DS1 and Dstar decays in the expected final state but K0 does not!");
+          }
+        }
+        auto indexMother = RecoDecay::getMother(particlesMc, vecDaughtersReso.back().template mcParticle_as<PParticles>(), Pdg::kDS1, true);
+        if (indexMother >= 0) {
+          auto particleMother = particlesMc.rawIteratorAt(indexMother);
+          motherPt = particleMother.pt();
+        }
+        fillHisto = true;
+      }
+    } 
+    else if constexpr (decChannel == DecayChannel::DplusV0) {
+      // Ds2Star → D+ K0 → (π+K-π+) K0s → (π+K-π+)(π+π-)
+      auto indexRec = RecoDecay::getMatchedMCRec(particlesMc, std::array{vecDaughtersReso[0], vecDaughtersReso[1], vecDaughtersReso[2], vecDaughtersReso[3], vecDaughtersReso[4]}, Pdg::kDS2Star, std::array{+kPiPlus, -kKPlus, +kPiPlus,  +kPiPlus, -kPiPlus}, true, &sign, 3);
+      if (indexRec > -1) {
+        // D+ → π+ K- π+
+        indexRec = RecoDecay::getMatchedMCRec(particlesMc, std::array{vecDaughtersReso[0], vecDaughtersReso[1], vecDaughtersReso[2]}, Pdg::kDPlus, std::array{+kPiPlus, -kKPlus, +kPiPlus}, true, &signDPlus, 1);
+        auto indexRecK0 = RecoDecay::getMatchedMCRec(particlesMc, std::array{vecDaughtersReso[3], vecDaughtersReso[4]}, kK0 , std::array{+kPiPlus, -kPiPlus}, true, &signV0, 2);
```

Comment:
Why do we check `kK0` and not `kK0Short`?
```suggestion
        auto indexRecK0 = RecoDecay::getMatchedMCRec<false, true>(particlesMc, std::array{vecDaughtersReso[3], vecDaughtersReso[4]}, kK0 , std::array{+kPiPlus, -kPiPlus}, true, &signV0, 2);
```

Reply:
Here specifically I think it should work either way, but elsewhere I can't use the K0s because of how signs are treated in the matching function